src/geometry/LinearPropagator.hh

```diff
@@ -21,18 +21,26 @@ class LinearPropagator
     using Initializer_t = GeoStateInitializer;
 
   public:
+    //! Output results
+    struct OutputType
+    {
+        real_type distance; //!< Distance traveled to min (input limit, boundary)
+        VolumeId  volume;   //!< Post-propagation volume
+    };
+
     // Construct from persistent and state data
     inline CELER_FUNCTION LinearPropagator(GeoTrackView& track);
 
-    // Move track by next_step(), which takes it to next volume boundary.
-    inline CELER_FUNCTION void operator()();
+    // Move track by next_step(), which takes it just across the next volume boundary.
+    inline CELER_FUNCTION OutputType operator()();
 
-    // Move track by a user-provided distance.
-    inline CELER_FUNCTION void operator()(real_type dist);
+    // Move track by a user-provided distance, or to the next boundary if distance is large enough.
+    // @return distance travelled along straight line
```

Comment:
Minor style change: `@return` is a doxygen comment that should be wherever the doxygen comment (`///`, `//!`, or `//@`etc) is defined: in this case, it should be in the `.i.hh`.